### DISCUSSION TITLE
Load/Unload Train When Longer Than Station

### DISCUSSION THREAD

**Discussion by Phelzier** [voxeltycoon/issues#479] [General]
**Is your feature request related to a problem? Please describe.**
When a train longer than the station is loading or unloading, the train loads all cars even though they are not in the station.

**Describe the solution you'd like**
In my opinion, the train should load the cars that are in the station and then move forward until the next empty car is at the front of the station.  Continue to load the relevant cars, repeat as necessary

**Describe alternatives you've considered**
Apparently it takes longer to load cars not in station, but I don't think cars should load when not in a station.  I figure in real life the train fills each car one by one using a flood loader.

**Additional context**

![bug2](https://user-images.githubusercontent.com/22050004/60390040-709bbc00-9ac5-11e9-8d8e-55e992e71a36.png)
.

**Comment by bigyihsuan** [comment]
I think what would help with this is dynamic station lengths, and not the fixed length that it is now.

**Comment by katepolak** [comment]
@bigyihsuan you can change station lengths though, refer to the hot key list in the bottom right of the screen when building a station to see how.

**Comment by katepolak** [comment]
@Phelzier currently when the train is longer it will load/unload slower. That is in my opinion better game design wise as the train slowly moving through the station as you suggested would be quite un-intuitive for new players, not to mention that it would be a lot of work to implement.

**Comment by katepolak** [comment]
@Phelzier Also please use the suggestion template for you future suggestions.

**Comment by RaptorSwire** [comment]
...imagine that instead of building station, you would DEFINE where the station segment is on the track. This way you would be able to build curved stations as they are often in real life ;)
